## Preparation

### Importing modules

In [114]:
import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
import matplotlib.pyplot as plt
import seaborn as sns

import os
print(os.listdir("../input"))

plt.style.use('ggplot')
%matplotlib inline

['athlete_events.csv', 'noc_regions.csv']


### Importing files

In [115]:
PATH = '../input/'
events = pd.read_csv(f'{PATH}athlete_events.csv')
regions = pd.read_csv(f'{PATH}noc_regions.csv')

### Exploration

In [116]:
events.head()

ID                      Name  ...                               Event  Medal
0   1                 A Dijiang  ...         Basketball Men's Basketball    NaN
1   2                  A Lamusi  ...        Judo Men's Extra-Lightweight    NaN
2   3       Gunnar Nielsen Aaby  ...             Football Men's Football    NaN
3   4      Edgar Lindenau Aabye  ...         Tug-Of-War Men's Tug-Of-War   Gold
4   5  Christine Jacoba Aaftink  ...    Speed Skating Women's 500 metres    NaN

[5 rows x 15 columns]

In [117]:
regions.head()

NOC       region                 notes
0  AFG  Afghanistan                   NaN
1  AHO      Curacao  Netherlands Antilles
2  ALB      Albania                   NaN
3  ALG      Algeria                   NaN
4  AND      Andorra                   NaN

Let's join both the tables on the 'NOC' column.

In [118]:
data = pd.merge(events, regions, how='left', on='NOC')

In [119]:
data.shape

(271116, 17)

In [120]:
data.columns

Index(['ID', 'Name', 'Sex', 'Age', 'Height', 'Weight', 'Team', 'NOC', 'Games',
       'Year', 'Season', 'City', 'Sport', 'Event', 'Medal', 'region', 'notes'],
      dtype='object')

## Participation statistics

### Country Participation over years

In [121]:
len(data.region.unique())

206

So, we have 206 unique countries who have participated in the Olympics over the years.

In [122]:
data.groupby(['Year','Season']).count()['region']

Year  Season
1896  Summer      380
1900  Summer     1936
1904  Summer     1301
1906  Summer     1733
1908  Summer     3101
1912  Summer     4038
1920  Summer     4292
1924  Summer     5233
      Winter      460
1928  Summer     4992
      Winter      582
1932  Summer     2969
      Winter      352
1936  Summer     6506
      Winter      895
1948  Summer     6404
      Winter     1075
1952  Summer     8263
      Winter     1088
1956  Summer     5075
      Winter     1307
1960  Summer     8114
      Winter     1116
1964  Summer     7702
      Winter     1778
1968  Summer     8582
      Winter     1891
1972  Summer    10292
      Winter     1655
1976  Summer     8637
      Winter     1861
1980  Summer     7191
      Winter     1746
1984  Summer     9440
      Winter     2134
1988  Summer    12010
      Winter     2639
1992  Summer    12940
      Winter     3436
1994  Winter     3160
1996  Summer    13745
1998  Winter     3605
2000  Summer    13798
2002  Winter     4109
2004  Summer    134

This shows that there is a disparity in how the summer and winter olympics were held in the same year before 1994, when they were separated to take place in gaps of 2 years. Let's separate them according to seasons.

In [123]:
summer_data = data[data.Season == 'Summer']
winter_data = data[data.Season == 'Winter']

Let's see the number of nations participating in the games in different seasons.

In [ ]:
summer_data.groupby('Year').region.nunique().plot.line(figsize=(20,8), label = 'summer')
winter_data.groupby('Year').region.nunique().plot.line(figsize=(20,8), label = 'winter')
plt.title('Participation over years', fontsize=14)
plt.legend(loc='best',fontsize=12)
